frontend/src/components/headers/DocumentHeader.tsx

```diff
@@ -84,7 +128,37 @@ function DocumentHeader() {
 							)}
 						</Paper>
 						<DownloadMenu />
+
+						<Stack alignItems="center">
+							<FormControl>
+								<FormContainer
+									defaultValues={{ title: documentTitle }}
+									onSuccess={handleUpdateDocumentTitle}
+								>
+									<Stack gap={4} alignItems="flex-end" flexDirection="row">
+										<TextFieldElement
+											variant="standard"
+											name="title"
+											label={documentTitle}
+											required
+											fullWidth
+											inputProps={{
+												maxLength: 255,
+											}}
+											onChange={handleDocumentTitleChange}
+											onFocus={handleFocus}
+										/>
+										{focused && (
+											<Button type="submit" variant="contained" size="large">
+												Update
+											</Button>
+										)}
+									</Stack>
+								</FormContainer>
+							</FormControl>
+						</Stack>
```

Comment:
When the user accesses the document in share mode, this feature should be disabled. We can check this information using the `shareRole` in `editorSlice`.